### PR TITLE
Bug/642656

### PR DESCRIPTION
**Pull request by jrburke**

### CODE REVIEW

File: web/dev/1/scripts/blade/Widget.js [hunk 1]
```diff
@@ -35,31 +35,71 @@ function (require,   object,     jig,     module) {
...
+                    asyncCreate.then(onFinishCreate);
+                    this.asyncCreate = asyncCreate;
+                } else {
+                    onFinishCreate();
                 }
             },
 
+            // poor man deferred thingy for now. Not very robust or resistant
+            // to tampering, but all the code is friendly here. Swap out
+            // with a real promise implementation at some point.
```
Comment (mixedpuppy) [inline web/dev/1/scripts/blade/Widget.js:75]: we should have a bug for this if there is a TODO

File: web/dev/1/scripts/accounts.js [hunk 2]
```diff
@@ -35,187 +35,19 @@ function (storage,   dispatch,   rdapi,   services) {
...
-          dispatch.pub('accountsChanged');
-        },
-
-        onChange: function (action) {
-          dispatch.sub('accountsChanged', action);
-        }
+  // Listen for chrome storage changes, and if it is for
+  // serviceCache, translate that into a higher level message
+  // understood by the web modules.
+  dispatch.sub('storeNotifyChange', function (data) {
```
Comment (mixedpuppy) [inline web/dev/1/scripts/accounts.js:48]: do we also need to listen for storeNotifyCleared?

File: web/dev/1/scripts/storage.js [hunk 3]
```diff
@@ -25,28 +25,85 @@
...
+      keyCallbacks.forEach(function (callback) {
+        callback(value);
+      });
+
+      delete callbacks[key];
+    }
+
+  });
+
+  dispatch.sub('storeNotifyChange', function (data) {
```
Comment (mixedpuppy) [inline web/dev/1/scripts/storage.js:94]: do we need to watch for storeNotifyCleared?

File: web/dev/1/scripts/blade/Widget.js (hunk 1 above)
Comment (jrburke) [inline web/dev/1/scripts/blade/Widget.js:75]: I'll update the comment per our discussion, it is not a security thing, more of a comment on how promises/deferreds can be constructed and not a real tampering/security issue.

**Pull request merged**